TAO/TAO_IDL/util/utl_global.cpp

```diff
@@ -1953,35 +1953,41 @@ namespace
   {
   public:
     explicit OldState (bool disable_output = false)
-        : old_filename_ (idl_global->filename ()),
+        : old_filename_ (idl_global->filename () ? new UTL_String (idl_global->filename (), true) : 0),
+          // need a copy because IDL_GlobalData::set_filename() destroys previous value
           old_lineno_ (idl_global->lineno ()),
           old_idl_src_file_ (idl_global->idl_src_file ()),
           disable_output_ (disable_output),
-          default_streambuf_ (0),
+          default_streambuf_ (ACE_DEFAULT_LOG_STREAM->rdbuf ()),
           flags_ (ACE_LOG_MSG->flags ())
       {
         idl_global->in_eval_ = true;
 
         idl_global->set_lineno (1);
         idl_global->set_filename (0);
 
-        // Name this pseudo-file "builtin-N"
-        static char buffer[64];
-        static unsigned n = 1;
-        ACE_OS::snprintf (&buffer[0], sizeof buffer, "builtin-%u", n++);
-        UTL_String utl_string (&buffer[0], true);
-        idl_global->idl_src_file (new UTL_String (&utl_string, true));
-        idl_global->set_filename (new UTL_String (&utl_string, true));
+        UTL_String filename = this->pseudo_filename ();
+        idl_global->idl_src_file (new UTL_String (&filename, true));
+        idl_global->set_filename (new UTL_String (&filename, true));
 
         if (disable_output_)
           {
-            default_streambuf_ = ACE_DEFAULT_LOG_STREAM->rdbuf ();
             ACE_DEFAULT_LOG_STREAM->rdbuf (0);
             ACE_LOG_MSG->clr_flags (ACE_Log_Msg::STDERR);
             ACE_LOG_MSG->clr_flags (ACE_LOG_MSG->flags ());
          }
       }
 
+      UTL_String pseudo_filename () const
+      {
+        // Name this pseudo-file "builtin-N"
+        static char buffer[64];
+        ACE_OS::snprintf (&buffer[0], sizeof buffer, "builtin-%u", pseudo_filename_counter_);
+        const UTL_String filename (&buffer[0], true);
+
+        return filename;
```

Comment:
Ups, UTL_String does not have a copy constructor, so we cannot return a UTL_String. On linux this works by accident, but with visual studio on windows we'll get an access violation because the filename object is "bogus".

I'll revert to how it was before the pseudo_filename() helper function.

Reply:
Everything works now